mysys/blackbox.h

```diff
@@ -0,0 +1,114 @@
+
+#pragma once
+
+#include <stdio.h>
+#include <cstring>
+#include <memory>
+#include <mutex>
+#include <vector>
+
+extern char *blackbox_filename;
+
+/** Stores log messages in a fixed size ringbuffer, which can be dumpeb by the
+ * crash detection code to provide additional debug information.
+ */
+class blackbox_logger {
+  using data_t = std::unique_ptr<std::vector<char>>;
```

Comment:
any reason why std::vector<char> is preferred to std::string?

Reply:
Only that the size of the vector is more explicit than the capacity of a string, but that's not that important, I changed it to string.